<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VoleMachineSimulator</class>
 <widget class="QMainWindow" name="VoleMachineSimulator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>480</y>
      <width>231</width>
      <height>61</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="addInstructionButton">
       <property name="text">
        <string>Add Instruction</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clearMemoryButton">
       <property name="text">
        <string>Clear Memory</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="Exit">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>500</y>
      <width>100</width>
      <height>29</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #4CAF50;      /* Green color /
    color: white;                   / Text color /
    border: none;                   / No border /
    padding: 10px 20px;             / Padding /
    border-radius: 5px;             / Rounded corners /
    font-size: 16px;                / Font size /
}

QPushButton:hover {
    background-color: #45a049;      / Darker green on hover /
}

QPushButton:pressed {
    background-color: #3e8e41;      / Even darker green when pressed */
}</string>
    </property>
    <property name="text">
     <string>Exit</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="decodeMessage">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>240</y>
      <width>281</width>
      <height>71</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="placeholderText">
     <string>Decoding Message</string>
    </property>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>40</y>
      <width>141</width>
      <height>331</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="RegistersLabel">
       <property name="text">
        <string>Registers</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="registersList"/>
     </item>
     <item>
      <widget class="QPushButton" name="clearRegistersButton">
       <property name="text">
        <string>Clear Registers</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>561</x>
      <y>50</y>
      <width>201</width>
      <height>171</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QLabel" name="PC_Label">
       <property name="text">
        <string>Program Counter</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="pcVeiwBox"/>
     </item>
     <item>
      <widget class="QPushButton" name="fetchButton">
       <property name="text">
        <string>fetch</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="decodeButton">
       <property name="text">
        <string>decode</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="executeButton">
       <property name="text">
        <string>execute</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>231</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="MainMemoryLabel">
       <property name="text">
        <string>Main Memory</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="memoryList">
       <property name="isWrapping" stdset="0">
        <bool>false</bool>
       </property>
       <property name="resizeMode">
        <enum>QListView::ResizeMode::Fixed</enum>
       </property>
       <property name="currentRow">
        <number>-1</number>
       </property>
       <property name="sortingEnabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="addInstructionLabel">
       <property name="text">
        <string>Add Instruction</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="addInstructionBox"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>390</y>
      <width>271</width>
      <height>131</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Screen</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTextEdit" name="screenBox">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menua">
    <property name="title">
     <string>Vole Machine Simulator</string>
    </property>
   </widget>
   <addaction name="menua"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>Exit</sender>
   <signal>clicked()</signal>
   <receiver>VoleMachineSimulator</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>687</x>
     <y>541</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>560</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
